rtos/TARGET_CORTEX/TOOLCHAIN_ARM_MICRO/mbed_boot_arm_micro.c

```diff
@@ -68,3 +68,20 @@ MBED_WEAK void __aeabi_assert(const char *expr, const char *file, int line)
 {
     mbed_assert_internal(expr, file, line);
 }
+
+// The below resolves the linker error generated by a bug in Arm Compiler 6
+// When building with MicroLib, the compiler inadvertently introduces the
+// _scanf_mbtowc symbol to the build. The code below provides a weak reference
+// for the missing symbol
+#include <stdio.h>
+typedef int ScanfReadRec;
+MBED_WEAK long int _scanf_mbtowc(
```

Comment:
Do we know what this function does? Basically what will fail for this setup.

Reply:
It converts multibyte charaters to wide characters. This function is not used in MicroLib which does not support wide characters or multibyte strings. The compiler inadvertently adds this symbol even if it is not used. There is an internal ticket being looked at by the compiler team which has acknowledged the bug.
The present commit essentially provides a stud for an unused functionality to satisfy the linker.

Reply:
Can we add here what version is affected (to know when we can remove it) , something along the lines <6.x.x is affected.

Reply:
> Can we add here what version is affected (to know when we can remove it) , something along the lines <6.x.x is affected.

6855093

Reply:
The info about the compiler versions affected has been last in the last commit.

Reply:
*lost, correct, I cant find the comment "Arm Compiler 6 version 6.12 and earlier versions are affected." here.

Reply:
It has 

> The info about the compiler versions affected has been last in the last commit.

It has now been restored. Thanks.